add-on/src/popup/browser-action/store.js

```diff
@@ -177,6 +179,11 @@ export default (state, emitter) => {
       }
       // console.dir('toggleSiteIntegrations', state)
       await browser.storage.local.set({ disabledOn, enabledOn })
+      await notifyOptionChange()
+      // notifyOptionsChange call is async, sends a message to background and
+      // waits for it to resolve. However, that doesnt work the
+      // same way in Chrome and FF. So we need to wait a bit before reloading.
+      await new Promise(resolve => setTimeout(resolve, 200))
```

Comment:
Is there a communication bus we can use to remove the need for this arbitrary wait time?

Reply:
@whizzzkid  I think this should be solved/moved to `sendMessageToSelf`, perhaps? there is a comment about discrepancy between Chrome/Firefox there already. 

We don't want random timeouts like that sprinkled in random places, will make debugging of future race conditions even harder.